quickwit/quickwit-config/src/source_config/mod.rs

```diff
@@ -355,6 +356,23 @@ pub enum PulsarSourceAuth {
     },
 }
 
+// Deserializing a string into an pulsar uri.
+fn pulsar_uri<'de, D>(deserializer: D) -> Result<String, D::Error>
+where D: Deserializer<'de> {
+    let uri: String = Deserialize::deserialize(deserializer)?;
+
+    let err_msg = format!(
+        "Invalid Pulsar uri provided, must be in the format of `pulsar://host:port/path`. Got: \
+         `{uri}`"
+    );
+
+    let _remaining = uri
```

Comment:
nitpick: an `if` condition would have been a bit clearer.

Reply:
AH yes, originally this was used but then I remembered IPV6 was allowed which broke everything :(

Shall fix!